preprocess_web/code/ravens_metadata_apps/preprocess_jobs/job_util.py

```diff
@@ -181,43 +169,68 @@ def retrieve_rows_json(job, **kwargs):
         # -------------------------------------------
         # Read partial file, set lines to skip, etc
         # -------------------------------------------
-        start_row_idx = start_row - 1    # e.g. if start_row is 1, skip nothing
-                                    # if start_row is 10, start on index
+        start_row_idx = start_row - 1  # e.g. if start_row is 1, skip nothing
+                                       # if start_row is 10, start on index
 
+        # ------------------------------------------
+        # Check the errors
+        # ------------------------------------------
+        error_message = []
+        job_data = job.get_metadata()
+        if job_data:
+            row_cnt = job_data[1]['dataset']['row_cnt']
+            print("row_cnt ", row_cnt)
+            if start_row > row_cnt:
+                err = 'The request was from %s rows but only %d rows were found, so default start rows = 1 is set'\
+                        % (start_row, row_cnt)
+                error_message.append(err)
+                start_row = 1
+                start_row_idx = start_row - 1
+            if num_rows > row_cnt:
```

Comment:
adjust to handle varying `start_row`; 
e.g. what if `start_row` + `num_rows` > row_cnt

Reply:
I have added the error but read_csv handles it and return the rows within the frame e.g. if start row = 3 num_rows = 20 and row_cnt= 15 then read_csv returns rows 3 to 15.